### PR TITLE
expressions: Support `bin-to-uuid` and `uuid-to-bin`

### LINKED ISSUE #20119: Support `bin-to-uuid` and `uuid-to-bin`
## Feature Request

**Is your feature request related to a problem? Please describe:**

Support new builtin functions in MySQL 8.0. Improve compatibility.

**Describe the feature you'd like:**

[uuid-to-bin](https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_uuid-to-bin) and [bin-to-uuid](https://dev.mysql.com/doc/refman/8.0/en/miscellaneous-functions.html#function_bin-to-uuid)

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by b41sh**
### What problem does this PR solve?

Issue Number: close #20119 

Problem Summary:
Support `bin-to-uuid` and `uuid-to-bin`

### What is changed and how it works?

What's Changed:
add `uuidToBinFunctionClass` and `binToUuidFunctionClass`

How it Works:

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 

- Support `bin-to-uuid` and `uuid-to-bin`

### CODE REVIEW

File: expression/builtin_miscellaneous.go [hunk 1]
```diff
@@ -1044,3 +1048,148 @@ type uuidShortFunctionClass struct {
...
+			return "", false, err
+		}
+	}
+	if flag != 0 {
+		swapBin := make([]byte, len(bin))
+		copy(swapBin[0:2], bin[6:8])
+		copy(swapBin[2:4], bin[4:6])
+		copy(swapBin[4:8], bin[0:4])
+		copy(swapBin[8:], bin[8:])
+		return string(swapBin), false, nil
```
Comment (qw4990) [inline expression/builtin_miscellaneous.go:1119]: @b41sh sorry for our late reply.
Can we modify `bin` here? Then we can:
```
buf := make([]byte, 4)
copy(bin, bin[0:4]
copy(bin[0:2], bin[6:8]
copy(bin[2:4], bin[4:6];
copy(bin[4:8], but)
return string(bin)
```

Comment (qw4990) [reply expression/builtin_miscellaneous.go:1119]: And how about creating a function for this logic? Then both the row-based and vectorized implementation can use it.

Comment (qw4990) [reply expression/builtin_miscellaneous.go:1119]: Almost LGTM, please address comments. @b41sh

Comment (b41sh) [reply expression/builtin_miscellaneous.go]: I add two functions `swapBinaryUUID` and `swapStringUUID` to handle UUID swapping
PTAL @qw4990

**Review by qw4990 — COMMENTED** [review]
LGTM

**Comment by qw4990** [comment]
PTAL @XuHuaiyu @SunRunAway

**Comment by b41sh** [comment]
PTAL @XuHuaiyu @wshwsh12

**Review by XuHuaiyu — APPROVED** [review]
LGTM

**Pull request merged**